File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/domain/Role.java
package org.jimmer.jimmerdatasouce.domain;

import org.babyfish.jimmer.sql.*;

import java.time.LocalDateTime;

/**
 * @author jhlz
 * @version x.x.x
 */
@Entity
@Table(name = "tb_role")
public interface Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long roleId();

    String roleCode();

    String roleName();

    LocalDateTime createDatetime();

    LocalDateTime updateDatetime();

    String remark();

    int status();
}

File: jimmer-example/quarkus-with-jimmer-kt/src/main/kotlin/org/jimmer/domain/Role.kt
package org.jimmer.domain

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * @author jhlz
 * @since 2024/4/17 13:48
 * @version x.x.x
 */
@Entity
@Table(name = "tb_role")
interface Role {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val roleId: Long

    val roleName: String
    val roleCode: String
    val createTime: LocalDateTime?
    val updateTime: LocalDateTime?
    val remark: String?

    /**
     * 状态0正常，1停用
     */
    val status: Int
}